### ISSUE TITLE
processed_source.path now nil in the autocorrect method

### ISSUE THREAD

**Issue by tas50** [rubocop-hq/rubocop#8264]
## Expected behavior

I realize the autocorrect behavior changed in 0.87, but I was evaluating a stepped upgrade where I went to 0.87 and then updated my nearly 200 cops to use the new format. All my specs pass after going from 0.86 -> 0.87.1 except for one cop that relies on `processed_source.path` within the autocorrect method. Previously this would return the file name, but now it returns nil. If this is intentional feel free to close this out, but this seemed like an odd break when everything else functioned as expected.

```ruby
          def autocorrect(_node)
            lambda do |_corrector|
              path = processed_source.path
              cb_name = File.basename(File.dirname(path))

              metadata = IO.read(path)
              IO.write(path, "name '#{cb_name}'\n" + metadata)
            end
          end
```

## Actual behavior

`processed_source.path` is now nil

## Steps to reproduce the problem

Access `processed_source.path` within the autocorrect method.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler. Here's an example:

```
$ [bundle exec] rubocop -V
0.87.1 (using Parser 2.7.1.4, rubocop-ast 0.1.0, running on ruby 2.7.1 x86_64-darwin19)
```

**Comment by tas50** [comment]
Scratch this. It's only impacting specs and not real world data.

**Comment by marcandre** [comment]
I was just about to write that I wasn't able to reproduce this 😆

**Comment by marcandre** [comment]
BTW: `expect_offense` accepts a `file` parameter which you should set to the path you want.
